routing_common/transit_serdes.hpp

```diff
@@ -110,10 +116,9 @@ class Deserializer
 
   void operator()(m2::PointD & p, char const * /* name */ = nullptr)
   {
-    m2::PointU pointU;
-    pointU.x = ReadVarUint<uint32_t, Source>(m_source);
-    pointU.y = ReadVarUint<uint32_t, Source>(m_source);
+    m2::PointU pointU = DecodeDelta(ReadVarUint<uint64_t, Source>(m_source), m_lastDecodedPoint);
```

Comment:
If you're going to encode/decode polylines with this method, I suggest you to try EncodePolyline/DecodePolyline.

Reply:
Yes. I'll try to use EncodePolyline/DecodePolyline and get rid of m_lastEncodedPoint/m_lastDecodedPoint and Reset method at Serializer/Deserializer.

 I can do that if I implement: void operator()(std::vector<m2::PointD> const & vs, char const * /* name */ = nullptr) at Serializer/Deserializer.

Reply:
I tried to used EncodePolyline/DecodePolyline. I didn't get significant profit in terms of section size but I get complex code:
```c++
  void Serializer::operator()(std::vector<m2::PointD> const & vs, char const * /* name */ = nullptr)
  {
    CHECK_LESS_OR_EQUAL(vs.size(), std::numeric_limits<uint64_t>::max(), ());
    WriteVarUint(m_sink, static_cast<uint64_t>(vs.size()));

    if (vs.empty())
      return;

    std::vector<m2::PointU> pointsU(vs.size());
    m2::PointU avrU = vs[0];
    m2::PointU maxU = vs[0];
    pointsU[0] = PointD2PointU(vs[0], POINT_COORD_BITS);;
    for (size_t i = 1; i < vs.size(); ++i)
    {
      m2::PointU const u = PointD2PointU(vs[i], POINT_COORD_BITS);
      pointsU[i] = u;
      avrU = avrU * (static_cast<double>(i) / (i + 1)) + u / static_cast<double>(i + 1);
      if (u.x > maxU.x)
        maxU.x = u.x;
      if (u.y > maxU.y)
        maxU.y = u.y;
    }

    WriteVarUint(m_sink, avrU.x);
    WriteVarUint(m_sink, avrU.y);
    WriteVarUint(m_sink, maxU.y);
    WriteVarUint(m_sink, maxU.y);

    std::vector<uint64_t> deltas;
    deltas.resize(vs.size());
    geo_coding::OutDeltasT deltasW(deltas);
    geo_coding::EncodePolyline(make_read_adapter(pointsU), avrU, maxU, deltasW);

    for (uint64_t d : deltas)
      WriteVarUint(m_sink, d);
  }

  void Deserializer::operator()(vector<m2::PointD> & vs, char const * /* name */ = nullptr)
  {
    auto const size = ReadVarUint<uint64_t, Source>(m_source);
    if (size == 0)
      return;

    m2::PointU avrU;
    avrU.x = ReadVarUint<uint32_t, Source>(m_source);
    avrU.y = ReadVarUint<uint32_t, Source>(m_source);
    m2::PointU maxU;
    maxU.x = ReadVarUint<uint32_t, Source>(m_source);
    maxU.y = ReadVarUint<uint32_t, Source>(m_source);

    std::vector<uint64_t> deltas;
    deltas.resize(size);
    for (size_t i = 0; i < size; ++i)
      deltas[i] = ReadVarUint<uint64_t, Source>(m_source);

    vector<m2::PointU> decodedPoints;
    decodedPoints.resize(size);

    geo_coding::OutPointsT decodedPointsA(decodedPoints);
    geo_coding::DecodePolyline(make_read_adapter(deltas), avrU, maxU, decodedPointsA);
    vs.resize(size);
    for (size_t i = 0; i < size; ++i)
      vs[i] = PointU2PointD(decodedPoints[i], POINT_COORD_BITS);
  }
```

I don't like to complexity. I did a special implementation for 
```c++
void Serializer::operator()(std::vector<m2::PointD> const & vs, char const * /* name */ = nullptr);
void Deserializer::operator()(vector<m2::PointD> & vs, char const * /* name */ = nullptr);
```
and get rid of `m_lastEncodedPoint` and m_lastDecodedPoint members at Serializer/Deserializer. For points at this method I use EncodeDelta/DecodeDelta.

Reply:
If you didn't get a significant profit, I agree that simpler implementation is better.